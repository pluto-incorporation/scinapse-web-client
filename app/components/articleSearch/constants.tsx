export const LOCAL_STORAGE_FILTERS = "l_s_filters";
export const PREVIOUS_FILTER = "prev_filters";
