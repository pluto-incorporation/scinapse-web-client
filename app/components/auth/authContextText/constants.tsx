export const positiveSignUpContext: { [key: string]: string } = {
  downloadPdf: "📜 To download PDF more, you need to be a Scinapse member.",
  citePaper: "🔍 To copy citation, you need to be a Scinapse member.",
  viewMorePDF: "📜 To view full text, you need to be a Scinapse member.",
  viewMorePDFUnlimited: "📜 To view full text, you need to be a Scinapse member.",
  addToCollection: "🔍 To add papers to collection, you need to be a Scinapse member.",
  clickRequestFullTextBtn: "📜 To request the full text of this paper, you need to be a Scinapse member.",
  query: "🔍 To search more, you need to be a Scinapse member.",
  paperShow: "📈 To view paper information more, you need to be a Scinapse member.",
  paperFromSearch: "📜 To view more papers, you need to be a Scinapse member.",
  queryLover: "💌 Hey Scinapse lover! Please help us to fully support your research by signing up.",
  nextPageFromSearch: "➡ To go to the next page, you need to be a Scinapse member.",
};
