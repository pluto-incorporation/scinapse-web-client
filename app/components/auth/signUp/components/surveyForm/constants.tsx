interface CheckedAnswer {
  name: string;
  order: number;
}

export interface RawQuestionType {
  question: string;
  answers: string[];
  random: boolean;
  type: string;
  description?: string;
}

interface QuestionType {
  question: string;
  random: boolean;
  checked: CheckedAnswer[];
}

export interface RawSurvey {
  surveyName: string;
  questions: RawQuestionType[];
}

export interface Survey {
  surveyName: string;
  questions: QuestionType[];
}
export const SCINAPSE_SURVEY_NAME = 'thinkAsScinapse';

export const SCINAPSE_SURVEY_QUESTIONS: RawSurvey = {
  surveyName: SCINAPSE_SURVEY_NAME,
  questions: [
    {
      question: 'What do you think as Scinapse?',
      answers: ['Publisher', 'PDF Provider', 'Search Engine'],
      random: true,
      type: 'radio',
    },
    { question: 'alpha', answers: ['a', 'b', 'c'], random: true, type: 'checkbox' },
  ],
};
