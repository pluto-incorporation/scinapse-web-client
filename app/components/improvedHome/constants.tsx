export const JOURNALS = [
  'nature',
  'science',
  'ieee',
  'cell',
  'acs',
  'aps',
  'lancet',
  'acm',
  'jama',
  'bmj',
  'pnas',
  'more-journals',
];

export const MOBILE_JOURNALS = ['nature', 'science', 'lancet', 'acm', 'ieee', 'cell', 'more-journal-mobile'];
