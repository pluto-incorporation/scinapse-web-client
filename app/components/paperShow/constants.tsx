export type RELATED_PAPERS = 'reference' | 'cited';
