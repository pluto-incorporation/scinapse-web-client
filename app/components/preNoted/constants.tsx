interface BlockedBubbleKeywords {
  verb: string;
  noun: string;
}

interface BlockedBubbleContextStructure {
  title: string;
  mainText: string;
}

const blockedBubbleVerbAndNoun: { [key: string]: BlockedBubbleKeywords } = {
  downloadPdf: { verb: "download", noun: "PDF" },
  clickRequestFullTextBtn: { verb: "request", noun: "paper" },
  viewMorePDF: { verb: "view", noun: "PDF" },
  addToCollection: { verb: "save to", noun: "Collection" },
};

export function getBlockedBubbleContext(userGroupName: string, buttonAction: string): BlockedBubbleContextStructure {
  const contextVerbAndNoun = blockedBubbleVerbAndNoun[buttonAction];
  switch (userGroupName) {
    case "1":
      return {
        title: `Do you want to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}?`,
        mainText: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}. It's free!`,
      };

    case "2":
      return {
        title: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}`,
        mainText: `When you join Scinapse, you can ${contextVerbAndNoun.verb} any ${contextVerbAndNoun.noun}.`,
      };

    case "3":
      const capitalizedVerb = contextVerbAndNoun.verb.charAt(0).toUpperCase + contextVerbAndNoun.verb.slice(1);
      return {
        title: `${capitalizedVerb} ${contextVerbAndNoun.noun} with a Scinapse account`,
        mainText: `Enjoy unlimited Scinapse with a Scinapse account.`,
      };
  }
  return { title: "", mainText: "" };
}
