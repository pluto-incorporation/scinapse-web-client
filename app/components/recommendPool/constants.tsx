export const BASED_ACTIVITY_COUNT_STORE_KEY = 'recommendPapersDialogCount';
export const ALREADY_VISITED_RECOMMEND_PAPERS = 'visited';
export const BASED_ACTIVITY_PAPER_IDS_FOR_NON_USER_KEY = 'b_a_p_ids';
