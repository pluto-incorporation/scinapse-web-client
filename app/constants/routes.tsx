export const HOME_PATH = "/";
export const SEARCH_RESULT_PATH = "/search";
export const AUTHOR_SEARCH_RESULT_PATH = "/search/authors";
export const AUTHOR_SHOW_PATH = "/authors/:authorId";
export const COLLECTION_LIST_PATH = "/users/:userId/collections";
export const AUTH_PATH = "/users";
export const PAPER_SHOW_PATH = "/papers/:paperId";
export const JOURNAL_SHOW_PATH = "/journals/:journalId";
export const COLLECTION_SHOW_PATH = "/collections/:collectionId";
export const TERMS_OF_SERVICE_PATH = "/terms-of-service";
export const PRIVACY_POLICY_PATH = "/privacy-policy";
export const ADMIN_PATH = "/admin";
