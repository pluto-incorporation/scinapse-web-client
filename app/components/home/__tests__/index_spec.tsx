import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter, Route } from 'react-router';
import { Provider } from 'react-redux';
import { generateMockStore } from '../../../__tests__/mockStore';
import { initialState } from '../../../reducers';
import HomeContainer from '..';
import { HOME_PATH } from '../../../constants/routes';

// TODO: react-test-renderer doesn't support React Hooks yet. It's merged to React source code, not deployed yet.
describe.skip('HomeContainer Component', () => {
  const mockStore = generateMockStore(initialState);

  beforeEach(() => {
    mockStore.clearActions();
  });

  it('should render content correctly', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore}>
          <MemoryRouter initialIndex={0} initialEntries={['/journals/2764552960']}>
            <Route path={HOME_PATH}>
              {/*Route is needed for providing match params*/}
              <HomeContainer />
            </Route>
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
