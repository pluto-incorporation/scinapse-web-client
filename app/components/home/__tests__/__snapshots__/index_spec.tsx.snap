// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer Component should render content correctly 1`] = `
<div
  className="articleSearchFormContainer"
>
  <h1
    style={
      Object {
        "display": "none",
      }
    }
  >
    Sci-napse | Academic search engine for paper
  </h1>
  <div
    className="searchFormInnerContainer"
  >
    <div
      className="searchFormContainer"
    >
      <div
        className="formWrapper"
      >
        <div
          className="searchTitle"
        >
          <span
            className="searchTitleText"
          >
             Do Research, Never Re-search
          </span>
          <img
            className="circleImage"
            src="https://assets.pluto.network/scinapse/circle%403x.png"
          />
          <img
            className="underlineImage"
            src="https://assets.pluto.network/scinapse/underline%403x.png"
          />
        </div>
        <div
          className="searchSubTitle"
        >
          Sci-napse is a free, nonprofit, Academic search engine 
          <br />
           for papers, serviced by
           
          <a
            className="plutoLink"
            href="https://pluto.network"
            target="_blank"
          >
            Pluto Network
          </a>
        </div>
        <div
          className="searchInputForm"
          tabIndex={0}
        >
          <div>
            <div
              className="inputWithListWrapper undefined"
              style={
                Object {
                  "backgroundColor": "white",
                  "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.07)",
                }
              }
            >
              <input
                autoFocus={true}
                onChange={[Function]}
                onKeyDown={[Function]}
                onSubmit={[Function]}
                placeholder="Search papers by title, author, doi or keyword"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "border": 0,
                    "borderRadius": "4px",
                    "display": "flex",
                    "height": "44px",
                    "overflow": "hidden",
                    "paddingLeft": "16px",
                    "width": "590px",
                  }
                }
                value=""
              />
              <span
                onClick={[Function]}
              >
                <i
                  className="icon searchIcon"
                />
              </span>
              <span
                onClick={[Function]}
              />
            </div>
            <ul
              className="suggestionList"
              style={
                Object {
                  "boxShadow": "0 1px 2px 0 #bbc2d0",
                }
              }
            />
          </div>
        </div>
        <div
          className="searchTryKeyword"
        />
      </div>
    </div>
    <div
      className="featureWrapper"
    >
      <div
        className="featureItem"
      >
        <div
          className="featureName"
        >
          Intuitive Feed
        </div>
        <div
          className="featureContents"
        >
          Quickly skim through the search results with major indices on the authors and the article.
        </div>
      </div>
      <div
        className="featureItem"
      >
        <div
          className="featureName"
        >
          Powered by community
        </div>
        <div
          className="featureContents"
        >
          Comments on the paper make it easy to find relevant papers for my research and comprehend them better
        </div>
      </div>
    </div>
    <div
      className="sourceVendorContainer"
    >
      <div
        className="sourceVendorSubtitle"
      >
        Metadata of papers comes from
      </div>
      <div
        className="sourceVendorWrapper"
      >
        <div
          className="sourceVendorItem"
        >
          <a
            href="https://aka.ms/msracad"
            target="_blank"
          >
            <img
              src="https://assets.pluto.network/scinapse/microsoft-research.png"
            />
          </a>
        </div>
        <div
          className="sourceVendorItem"
        >
          <a
            href="https://www.semanticscholar.org/"
            target="_blank"
          >
            <img
              src="https://assets.pluto.network/scinapse/semantic-scholar%402x.png"
            />
          </a>
        </div>
        <div
          className="sourceVendorItem"
        >
          <a
            href="https://www.springernature.com/gp/"
            target="_blank"
          >
            <img
              src="https://assets.pluto.network/scinapse/springernature%402x.png"
            />
          </a>
        </div>
        <div
          className="sourceVendorItem"
        >
          <a
            href="https://www.ncbi.nlm.nih.gov/pubmed/"
            target="_blank"
          >
            <img
              src="https://assets.pluto.network/scinapse/pubmed%402x.png"
            />
          </a>
        </div>
      </div>
    </div>
    <footer
      className="footerContainer"
      style={Object {}}
    >
      <div
        className="title"
      >
        <div>
          Â© 2019 Pluto Network. All rights reserved
        </div>
      </div>
      <div
        className="rightBox"
      >
        <a
          className="link"
          href="https://www.facebook.com/PlutoNetwork/"
          onClick={[Function]}
          target="_blank"
        >
          Facebook
        </a>
        <a
          className="link"
          href="https://twitter.com/pluto_network"
          onClick={[Function]}
          target="_blank"
        >
          Twitter
        </a>
        <a
          className="link"
          href="https://medium.com/pluto-network"
          onClick={[Function]}
          target="_blank"
        >
          Blog
        </a>
        <a
          className="link"
          href="https://pluto.network"
          onClick={[Function]}
          target="_blank"
        >
          About us
        </a>
        <a
          className="link"
          href="mailto:team@pluto.network"
          onClick={[Function]}
          target="_blank"
        >
          Contact us
        </a>
        <a
          className="link"
          href="https://www.notion.so/pluto/Frequently-Asked-Questions-4b4af58220aa4e00a4dabd998206325c"
          onClick={[Function]}
          target="_blank"
        >
          FAQ
        </a>
        <a
          className="termsOfServiceLink"
          href="https://scinapse.io/terms-of-service"
        >
          Terms of service
        </a>
      </div>
    </footer>
  </div>
</div>
`;
